### PR TITLE
refactor: execute insert with select in query engine

### LINKED ISSUE #760: Supports INSERT INTO SELECT statement
### What type of enhancement is this?

API improvement

### What does the enhancement do?

Using it to copy data from one table and inserts it into another table, such as:

```
INSERT INTO table2 SELECT * FROM table1 WHERE condition;
```

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Insert with select is now executed in query engine. Also insert with select is made possible in distributed mode.

However, plain insert (insert with values) is still executed in the form of statement, directly.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

#1010 

close #760

### CODE REVIEW

File: src/datanode/src/tests/instance_test.rs [hunk 1]
```diff
@@ -185,67 +185,6 @@ async fn test_execute_insert() {
     assert!(matches!(output, Output::AffectedRows(2)));
 }
 
-#[tokio::test(flavor = "multi_thread")]
-async fn test_execute_insert_by_select() {
```
Comment (MichaelScofield) [inline src/datanode/src/tests/instance_test.rs:189]: This test is moved to sqlness case.

**Review by sunng87 — APPROVED** [review]
LGTM

File: src/datanode/src/tests/instance_test.rs (hunk 1 above)
Comment (killme2008) [reply src/datanode/src/tests/instance_test.rs:191]: I think we should keep this test, it's part of the integration test. The sqlness test is a black box test and the developer may not run it while developing.

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**